File: Java/Easy/0001-two-sum.java
import java.util.HashMap;

class TwoSum {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int[] ans = new int[2];
        int compliment;
        for (int i = 0; i < nums.length; i++) {
            compliment = target - nums[i];
            if (map.containsKey(compliment)) {
                ans[0] = map.get(compliment);
                ans[1] = i;
            } else {
                map.put(nums[i], i);
            }
        }
        return ans;
    }
}

File: Python/Easy/0001-two-sum.py
class TwoSum:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        res = []
        for index, num in enumerate(nums):
            compliment = target - num
            if compliment in seen:
                return [seen[compliment], index]
            seen[num] = index
        return res
